code/drug_synergy/02-drug_synergy_score_calc.R

```diff
@@ -0,0 +1,135 @@
+# Author: Run Jin
+#
+# Obtain Drug Target of Interest for each qSig output (comparing to GTEx, PBTA all or PBTA HGG)
+
+suppressPackageStartupMessages(library("optparse"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("igraph"))
+
+#### Define Directories ----------------------------------------------------------
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+analysis_dir <- file.path(root_dir, "code", "drug_synergy")
+results_dir <- file.path(analysis_dir, "results", "synergy_score")
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+
+source(file.path(analysis_dir, "utils/get_synergy_score.R"))
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-g","--gtex_mapped"),type="character",
+              help="Output file for drug-mapped gtex qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-p","--pbta_mapped"),type="character",
+              help="Output file for drug-mapped pbta qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-h","--pbta_hgg_mapped"),type="character",
+              help="Output file for drug-mapped pbta hgg subsetted subnetworks (.tsv) "),
+  make_option(c("-f","--subnetwork"),type="character",
+               help="File for subnetworks of module of interest (.tsv) "),
+  make_option(c("-b","--output_path"),type="character",
+              help="Path for results output ")
+)
+opt <- parse_args(OptionParser(option_list=option_list,add_help_option = FALSE))
+output_path <- opt$output_path 
+
+#### Read in files necessary for analyses -----------------------------------
+gtex_qSig_subnet_mapped <- readr::read_tsv(opt$gtex_mapped) %>% mutate(comparison = "gtex_qSig")
+pbta_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_mapped) %>% mutate(comparison = "pbta_qSig")
+pbta_hgg_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_hgg_mapped) %>% mutate(comparison = "pbta_hgg_qSig")
+
+subnetwork <- readr::read_tsv(opt$subnetwork)
+
+#### Synergy Score calculated ----------------------------------------------------------
+# following steps adapted from SynGeNet Publication [@doi:10.1038/s41540-019-0085-4]
+# Supplementary information downloadable as `SynGeNet.R`
+
+#### SubNetwork Generation ---------------------------------------------------------------
+subnetwork_graphed <- subnetwork %>% select(Gene1, Gene2) %>% as.matrix() 
+
+  # Induce background network
+subnetwork_graphed <- graph.edgelist(subnetwork_graphed)
```

Comment:
with the subnetwork file you are reading in, would this background graph step run on every network module positively associated with the patient of interest (e.g. if there are multiple modules positively associated with the patient of interest)?

Reply:
Currently I chose the network that has the highest positive score so this would only be one network. Do we want to do all networks?

Reply:
Yes, expanding to all subnetworks would be the goal. The synergy output files you generate (e.g. `gtex_qSig_synergy_score.tsv`, etc) are nicely formatted, and adding the module number (e.g. `module` column with `Mx` designation) as an additional field would be informative. The final file could then continue to be ordered by synergy score in decreasing order as you have it.

Reply:
Sure! Will do that :)